pkg/proxy/apis/config/types.go

```diff
@@ -190,24 +187,20 @@ type KubeProxyConfiguration struct {
 	DetectLocal DetectLocalConfiguration
 }
 
-// ProxyMode represents modes used by the Kubernetes proxy server. Currently, three modes of proxy are available in
-// Linux platform: 'userspace' (older, going to be EOL), 'iptables' (newer, faster), 'ipvs'(newest, better in performance
-// and scalability).
+// ProxyMode represents modes used by the Kubernetes proxy server.
 //
-// Two modes of proxy are available in Windows platform: 'userspace'(older, stable) and 'kernelspace' (newer, faster).
+// Currently, two modes of proxy are available in Linux platform: 'iptables' and 'ipvs'.
+// One mode of proxy is available in Windows platform: 'kernelspace'.
 //
 // In Linux platform, if proxy mode is blank, use the best-available proxy (currently iptables, but may change in the
 // future). If the iptables proxy is selected, regardless of how, but the system's kernel or iptables versions are
-// insufficient, this always falls back to the userspace proxy. IPVS mode will be enabled when proxy mode is set to 'ipvs',
-// and the fall back path is firstly iptables and then userspace.
+// insufficient, kube-proxy will fail. IPVS mode will be enabled when proxy mode is set to 'ipvs'.
 //
-// In Windows platform, if proxy mode is blank, use the best-available proxy (currently userspace, but may change in the
-// future). If winkernel proxy is selected, regardless of how, but the Windows kernel can't support this mode of proxy,
-// this always falls back to the userspace proxy.
+// In Windows platform, if proxy mode is blank, use the best-available proxy (currently winkernel)
+// If winkernel proxy is selected, but the Windows kernel can't support this mode of proxy, kube-proxy will fail.
```

Comment:
This can all be simplified even more. And fixed up a bit too.

```
If the proxy mode is unspecified, the best-available proxy mode will be used (currently this
is `iptables` on Linux and `kernelspace` on Windows). If the selected proxy mode cannot be
used (due to lack of kernel support, missing userspace components, etc) then kube-proxy
will exit with an error.
```